homeassistant/components/switch/netio.py

```diff
@@ -0,0 +1,199 @@
+        varname..'='..tostring(devices.system[varname]):gsub(" ","|"))
+end end
+
+local qs = table.concat(output, '&')
+local url = string.format('http://%s%s?%s', address, path, qs)
+devices.system.CustomCGI{url=url}
+```
+
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.netio/
+
+"""
+import logging
+import socket
+from homeassistant import util
+from datetime import timedelta
+from pynetio import Netio
+from collections import namedtuple
+from homeassistant.const import *
+from homeassistant.helpers import validate_config
+from homeassistant.components.switch import SwitchDevice, \
+    ATTR_CURRENT_POWER_W, ATTR_TOTAL_CONSUMPTION_KWH
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['http']
+REQUIREMENTS = ['pynetio>=0.1.5.1']
+DEFAULT_USERNAME = 'admin'
+DEFAULT_PORT = 1234
+REQ_CONF = [CONF_HOST, CONF_PORTS]
+URL_API_NETIO_EP = "/api/netio"
+
+device = namedtuple('device', ['netio', 'entities'])
+DEVICES = {}
+ATTR_START_DATE = 'Start date'
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """ Configure the netio linl """
+
+    if validate_config({"conf": config}, {"conf": [CONF_PORTS,
+                                                   CONF_HOST]}, _LOGGER):
+        if not config[CONF_HOST] in DEVICES:
+            try:
+                dev = Netio(config[CONF_HOST],
+                            config.get(CONF_PORT, DEFAULT_PORT),
+                            config.get(CONF_USERNAME, DEFAULT_USERNAME),
+                            config.get(CONF_PASSWORD, DEFAULT_USERNAME))
+                DEVICES[config[CONF_HOST]] = device(dev, [])
+            except:
+                _LOGGER.error('Cannot connect to %s' % config[CONF_HOST])
+
+        if DEVICES.get(config[CONF_HOST]):
+            for key in config[CONF_PORTS]:
+                switch = NetioSwitch(DEVICES[config[CONF_HOST]].netio, key,
+                                     config[CONF_PORTS][key])
+                add_devices_callback([switch])
+                DEVICES[config[CONF_HOST]].entities.append(switch)
+
+    hass.http.register_path('GET', URL_API_NETIO_EP, _got_push)
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, dispose)
+    return True
+
+
+def _got_push(handler, path_match, data):
+    """ To handle updates from http GET """
+
+    host = socket.gethostbyaddr(handler.client_address[0])[0].split('.')[0]
+    states, consumptions, cumulatedConsumptions, startDates = [], [], [], []
+
+    for i in range(1, 5):
+        states.append(
+            True if data.get('output%d_state' % i) == STATE_ON else False)
+        consumptions.append(float(data.get('output%d_consumption' % i, 0)))
+        cumulatedConsumptions.append(
+            float(data.get('output%d_cumulatedConsumption' % i, 0)) / 1000)
+        startDates.append(data.get('output%d_consumptionStart' % i, ""))
```

Comment:
Since the `output%d_` prefix is used quite a few times, it might be better to calculate the prefix once and then just do string concatenations after.